tests/js/client/restart/test-restart-replication2-view-recovery-cluster.js

```diff
@@ -0,0 +1,545 @@
+          if (value.response.local.releaseIndex <= commitIndexAfterCompaction) {
+            return Error(`Participant ${pid} cannot compact enough, status: ${JSON.stringify(status)}, ` +
+              `log contents: ${JSON.stringify(logContents)}`);
+          }
+        }
+        return true;
+      });
+
+      // Trigger compaction intentionally. We aim to clear a prefix of the log here, removing the CreateShard entries.
+      // This way, the recovery procedure will be forced to handle entries referring to an unknown shard.
+      log.compact();
+    };
+
+
+    // This is not too impartant, we try to get into an
+    // "empty" and "filled" log with this.
+    // However the filled one is the problemeatic one.
+    // So this was more to proof that empty log is not a problem
+    for (const l of logs) {
+      // waitForLogToBeCompacted(l);
+    }
+  };
+
+  const insertDummyData = (number) => {
+    // NOTE: This is on purpose single calls.
+    // We need those to be different entries in the replicated log
+    // Otherwise we would only have a single entry for the batch
+    for (let i = 0; i < number; ++i) {
+      collection.save(genDoc());
+    }
+  };
+
+  const insertDummyUpdates = (number, updates) => {
+    // NOTE: This is on purpose single calls.
+    // We need those to be different entries in the replicated log
+    // Otherwise we would only have a single entry for the batch
+    const docs = [];
+    for (let i = 0; i < number; ++i) {
+      const doc = genDoc();
+      const {_key} = collection.save(doc);
+      docs.push({doc, _key});
+    }
+    for (let i = 0; i < updates; ++i) {
+      for (const {_key, doc} of docs) {
+        collection.update(_key, doc);
+      }
+    }
+  };
+
+  const insertDummyReplaces = (number, updates) => {
+    // NOTE: This is on purpose single calls.
+    // We need those to be different entries in the replicated log
+    // Otherwise we would only have a single entry for the batch
+    const docs = [];
+    for (let i = 0; i < number; ++i) {
+      const doc = genDoc();
+      const {_key} = collection.save(doc);
+      docs.push({doc, _key});
+    }
+    for (let i = 0; i < updates; ++i) {
+      for (const {_key, doc} of docs) {
+        collection.replace(_key, doc);
+      }
+    }
+  };
+
+  const assertViewInSync = () => {
+    const result = db._query(`FOR doc IN ${viewName} SEARCH doc.c >= 0 OPTIONS {waitForSync: true} RETURN doc._key`).toArray();
+    const expectedResult = db._query(`FOR doc IN ${collection.name()} FILTER doc.c >= 0 RETURN doc._key`).toArray();
+
+    const uniqView = new Set(result);
+    const uniqCol = new Set(expectedResult);
+    if (result.length !== expectedResult.length) {
+      {
+        const dh = require("@arangodb/testutils/document-state-helper");
+        let intermediateCommitEntries = dh.getDocumentEntries(dh.mergeLogs(logs), "Insert");
+        require("console").error(JSON.stringify(intermediateCommitEntries));
```

Comment:
Note to Reviewer:
The console output here is for more precise error reporting.
Therefore I would like to keep it